ver 0.0.2 fixed to build on ARM

h-denpo opened the pull request:
gotcha(ver0.0.2) install fail the following message:

We also need to patch gotcha(ver0.0.2).
```
 >> 50    /tmp/denpo/spack-stage/spack-stage-gotcha-0.0.2-67l4fo2kvb2gd64u4m5owx3mwxayzhom/spack-src/src/libc_wrappers.
           c:354:21: error: 'SYS_open' undeclared (first use in this function); did you mean 'SYS_openat'?
     51        result = syscall(SYS_open, pathname, flags, mode);
     52                         ^~~~~~~~
     53                         SYS_openat
     54    /tmp/denpo/spack-stage/spack-stage-gotcha-0.0.2-67l4fo2kvb2gd64u4m5owx3mwxayzhom/spack-src/src/libc_wrappers.
           c:354:21: note: each undeclared identifier is reported only once for each function it appears in
  >> 55    make[2]: *** [src/CMakeFiles/gotcha.dir/build.make:125: src/CMakeFiles/gotcha.dir/libc_wrappers.c.o] Error 1
     56    make[2]: *** Waiting for unfinished jobs....
     57    make[2]: Leaving directory '/tmp/denpo/spack-stage/spack-stage-gotcha-0.0.2-67l4fo2kvb2gd64u4m5owx3mwxayzhom/
           spack-build'
  >> 58    make[1]: *** [CMakeFiles/Makefile2:201: src/CMakeFiles/gotcha.dir/all] Error 2
     59    make[1]: Leaving directory '/tmp/denpo/spack-stage/spack-stage-gotcha-0.0.2-67l4fo2kvb2gd64u4m5owx3mwxayzhom/
           spack-build'
     60    make: *** [Makefile:153: all] Error 2
```

h-denpo committed:
ver 0.0.2 fixed to build on ARM

alalazo commented on var/spack/repos/builtin/packages/gotcha/package.py:
```diff
@@ -21,6 +21,7 @@ class Gotcha(CMakePackage):
 
     variant('test', default=False, description='Build tests for Gotcha')
     patch('arm.patch', when='@1.0.2')
+    patch('arm.patch', when='@0.0.2')
```
The fix for this issue is at https://github.com/LLNL/GOTCHA/commit/e82b4a1ecb634075d8f5334b796c888c86da0427 and it seems different from this patch. I suspect that the patched version will only work on aarch64 and not on other archs. Can you:
1. Remove the file patch from Spack repo
2. Use https://github.com/LLNL/GOTCHA/commit/e82b4a1ecb634075d8f5334b796c888c86da0427.patch as the url for the patch

?

h-denpo replied:
I agree with you.
I will make the changes as you advise.

h-denpo committed:
arm.patch removed
modify: use https://github.com/LLNL/GOTCHA/commit/e82b4a1ecb634075d8f5334b796c888c86da0427.patch as the url for the patch

h-denpo committed:
modify patch read from git

alalazo reviewed (approved):
Thanks @h-denpo !
